homeassistant/components/point/__init__.py

```diff
@@ -237,7 +241,7 @@ def __str__(self):
         _LOGGER.debug('Created device %s', self)
         self._async_unsub_dispatcher_connect = async_dispatcher_connect(
             self.hass, SIGNAL_UPDATE_ENTITY, self._update_callback)
-        self._update_callback()
+        await self.hass.async_add_executor_job(self._update_callback)
```

Comment:
The `_update_callback` should be run in the event loop, since it's decorated with `@callback`. Rework the whole part of that callback.

Reply:
I'm not sure what you mean, do you mean that this part should be updated? 

https://github.com/home-assistant/home-assistant/blob/30c4c2a1efd4c321a1d9b4626d1cb5b75dd872a5/homeassistant/components/point/sensor.py#L54

This part does IO, but I'm not sure how to make that async (without re-writing the pypoint library which is on my todo).
https://github.com/home-assistant/home-assistant/blob/30c4c2a1efd4c321a1d9b4626d1cb5b75dd872a5/homeassistant/components/point/sensor.py#L58

Reply:
Of you make the update callback a coroutine instead, we can await the executor inside the update callback.